### PR TITLE
VS.Vim.Buffer: Fix E11 error

### PR DESCRIPTION
**Pull request by mityu**
E11 error is raised when ensure() is called in cmdwin because `:badd` command isn't available in cmdwin.

### How to reproduce
`vim -u NONE -N -S /path/to/following/script.vim`
```vim
if !isdirectory('/tmp/vital-vs')
  !git clone --depth=1 https://github.com/hrsh7th/vim-vital-vs /tmp/vital-vs
endif

source /tmp/vital-vs/autoload/vital/__vital__/VS/Vim/Buffer.vim
let g:Ensure = function(printf("\<SNR>%d_ensure",
      \ matchstr(execute('filter Buffer.vim scriptnames'), '\d\+')))

try
  execute "normal! q:\<Cmd>call call(g:Ensure, ['test-vital-vs://newbuf'])\<CR>"
catch
  echomsg v:throwpoint
  echomsg v:exception
endtry
```

output:
```
command line..script /Users/mityu/baddrepro.vim[11]..function <SNR>2_ensure, line 5
Vim(badd):E11: Invalid in command-line window; <CR> executes, CTRL-C quits
```

### Solution
Prefer using `bufadd()` function to `:badd` command.

### Additional context
```vim
function! TestBadd()
  let buf1 = bufadd('newbuf1')
  call setbufvar(buf1, '&buflisted', 1)

  badd newbuf2
  let buf2 = bufnr('newbuf2')

  let buf1info = getbufinfo(buf1)[0]
  let buf2info = getbufinfo(buf2)[0]

  let diffs = []
  for [k, v1] in items(buf1info)
    let v2 = buf2info[k]
    if v1 != v2
      call add(diffs, printf('%s: (%s, %s)', k, v1, v2))
    endif
  endfor

  let msg = "Diffs:\n"
  for d in diffs
    let msg .= '  ' . d . "\n"
  endfor
  return msg
endfunction
echomsg TestBadd()
" Diffs:
"  bufnr: (2, 3)
"  name: (/Users/mityu/newbuf1, /Users/mityu/newbuf2)
```

As the script above, it seems that this script
```vim
let bufnr = bufadd(bufname)
call setbufvar(bufnr, '&buflisted', 1)
```
can alternate `:badd` command, but I'm not sure this is actually right.
Please tell me if there're some problems.

### CODE REVIEW

**Comment by mityu** [comment]
### More additional context

Since there're two implementation for `Buffer.load()` depending on `bufload()` existence, I also gave two implementation for `Buffer.badd()` depending on `bufadd()` existence. (helpful.vim says that `bufload()` function and `bufadd()` function are added in same patch.)


|`:HelpfulVersion`|output|
|:-|:-|
|`bufload()`|`bufload() \| vim: +v8.1.1610 \| neovim: +v0.4.0`|
|`bufadd()`|`bufadd() \| vim: +v8.1.1610 \| neovim: +v0.4.0`|
|`setbufvar()`|`setbufvar() \| vim: +v7.0001 \| neovim: +v0.1.0`|

File: autoload/vital/__vital__/VS/Vim/Buffer.vim [hunk 1]
```diff
@@ -42,11 +42,25 @@ function! s:ensure(expr) abort
...
-    badd `=a:expr`
+    call s:badd(a:expr)
   endif
   return bufnr(a:expr)
 endfunction
 
+"
+" badd
+"
+if exists('*bufadd')
```
Comment (hrsh7th) [inline autoload/vital/__vital__/VS/Vim/Buffer.vim:53]: I think it should be `add` via considering other method name.

**Comment by hrsh7th** [comment]
Also looks good. Thank you!

File: autoload/vital/__vital__/VS/Vim/Buffer.vim (hunk 1 above)
Comment (mityu) [reply autoload/vital/__vital__/VS/Vim/Buffer.vim:53]: Renamed.

**Comment by mityu** [comment]
Thank you for your review. Renaming method have done.

**Comment by hrsh7th** [comment]
LGTM. Thank you~.

**Comment by hrsh7th** [comment]
@mityu Oh. Sorry.... CI failed on some of the vim/nvim versions. Could you check this? (I also check it)

**Comment by hrsh7th** [comment]
I can't understand why lnum is not to be the same...

**Comment by hrsh7th** [comment]
It seems better to simplify the test. I don't think this difference is a big issue.

**Comment by mityu** [comment]
Oh, many CIs are failing...
I'm afraid but I'm very sleepy now, so I'll look into this tomorrow.

**Comment by mityu** [comment]
> I can't understand why lnum is not to be the same...

https://github.com/vim/vim/commit/e974fa7b2b98e53a59d5ef16a6a1291f12a32e33
I found this patch changed the `:badd`'s behavior, came to set lnum to 0. This is the reason why tests on old Vim/Neovim fails.

> It seems better to simplify the test. I don't think this difference is a big issue.

OK, I'll update the test.
Actually, `:h getbufinfo()` says that lnum is not available when the buffer has never been displayed.  I don't think the difference is a big issue neither.

**Comment by mityu** [comment]
Updated the test.

**Comment by mityu** [comment]
I'm looking into the other problem, please wait for a minutes..

**Comment by mityu** [comment]
Fix the other problem have done.

**Comment by hrsh7th** [comment]
The failing tests seems CI side problem. It looks good! Merging it.

**Comment by mityu** [comment]
Thank you.

**Pull request merged**